packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,175 @@ function WorklistDataGrid(props) {
     rows,
     columnHeaderHeight,
     onColumnSelect,
+    onCellSelect,
+    onRowSelect,
+    onRowSelectAll,
+    onClearSelectedRows,
+    onDisableSelectableRows,
+    hasSelectableRows,
     intl,
+    rowHeaderIndex,
   } = props;
 
+  const grid = useRef();
   const focusedRow = useRef(0);
   const focusedCol = useRef(0);
-  const grid = useRef();
+  const ariaLiveMsg = useRef();
+
+  const [currentSelectedCell, setCurrentSelectedCell] = useState(null);
+  const displayedColumns = (hasSelectableRows ? [WorklistDataGridUtils.ROW_SELECTION_COLUMN] : []).concat(columns);
+
+  const theme = useContext(ThemeContext);
+
+  const isRowSelectionCell = (columnIndex) => (
+    hasSelectableRows && columnIndex < displayedColumns.length && displayedColumns[columnIndex].id === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id
+  );
 
-  const gridRef = useCallback((node) => {
-    grid.current = node;
-    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+  const isCellSelected = (rowId, columnId) => (currentSelectedCell && currentSelectedCell.rowId === rowId && currentSelectedCell.columnId === columnId);
+
+  const setFocus = (rowIndex, colIndex, makeActiveElement) => {
+    let focusedCell = grid.current.rows[rowIndex].cells[colIndex];
+    if (isRowSelectionCell(colIndex) && focusedCell.firstChild) {
+      focusedCell = focusedCell.firstChild;
+    }
     focusedCell.tabIndex = 0;
-  }, []);
+    if (makeActiveElement && focusedCell.focus) {
+      focusedCell.focus();
+    }
+  };
 
-  const handleKeyDown = (event) => {
-    const currentFocusedRow = event.target.parentElement.rowIndex;
-    const currentFocusedCol = event.target.cellIndex;
+  const removeTabStop = (rowIndex, colIndex) => {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.firstChild) {
+      // Also remove tabstop from the checkbox in the case of a row selection cell.
+      cell.firstChild.tabIndex = -1;
+    }
```

Comment:
How can we be certain that the checkbox is the first child?  This makes any change to that cell structure break this logic.  Does the input field need to have a tabIndex?  Can't the RowSelectionCell handle the click and key events properly?

Reply:
The composition of the RowSelectionCell is internal and cannot be affected by the consumer so this constraint seemed reasonable. That said, I see your point and made the change we discussed to go after the child input element itself. 
Re, the RowSelectionCell itself handling it, this is about navigation and involves more than one cell so this should be done at the grid level. If we want to refine further, we can address in the row selection story.